src/Campaigns/resources/admin/components/CampaignDetailsPage/index.tsx

```diff
@@ -1,214 +1,165 @@
+                .then((response: Campaign) => {
+                    setIsSaving(null);
+                    reset(response);
+                })
+                .catch((response: any) => {
+                    setIsSaving(null);
+                    //todo: add error handling
+                    console.log(response);
+                });
         }
     };
 
+    if (!hasResolved) {
+        return (
+            <div className={styles.loadingContainer}>
+                <div className={styles.loadingContainerContent}>
+                    <GiveSpinner />
+                    <div className={styles.loadingContainerContentText}>
+                        {__('Loading campaign...', 'give')}
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
     return (
         <FormProvider {...methods}>
             <form onSubmit={handleSubmit(onSubmit)}>
-                <article className={styles.page}>
+                <article className={`interface-interface-skeleton__content ${styles.page}`}>
                     <header className={styles.pageHeader}>
                         <div className={styles.breadcrumb}>
-                            <a href={`${adminUrl}edit.php?post_type=give_forms&page=give-campaigns`}>
+                            <a href={`${window.GiveCampaignDetails.adminUrl}edit.php?post_type=give_forms&page=give-campaigns`}>
                                 {__('Campaigns', 'give')}
                             </a>
                             {' > '}
-                            <span>{campaign.properties.title}</span>
+                            <span>{campaign.title}</span>
                         </div>
                         <div className={styles.flexContainer}>
                             <div className={styles.flexRow}>
-                                <h1 className={styles.pageTitle}>{campaign.properties.title}</h1>
+                                <h1 className={styles.pageTitle}>{campaign.title}</h1>
                                 <span
                                     className={cx(
                                         styles.status,
-                                        campaign.properties.status === 'draft'
+                                        campaign.status === 'draft'
                                             ? styles.draftStatus
-                                            : styles.activeStatus
+                                            : styles.activeStatus,
                                     )}
                                 >
-                                    {campaign.properties.status}
+                                    {campaign.status}
                                 </span>
                             </div>
 
                             <div className={styles.flexRow}>
-                                {campaign.properties.status === 'draft' && (
+                                {campaign.status === 'active' && (
```

Comment:
The "Save as draft" button is not being displayed when the campaign is draft, only when it's active. So, if I want to edit the settings of the campaign without publishing it, it isn't possible as you can see on the attached screenshot. I think we need to display this button only when the campaign has the draft status and maybe a _"Convert to draft"_ button when it's active.

![image](https://github.com/user-attachments/assets/5922f94b-8483-4eea-a221-45d9f7d3e320)

Reply:
Good catch! Resolved: https://github.com/impress-org/givewp/pull/7547/commits/b41e8e9012f2079b632dd98cd516eba4e30817b6